src/main/java/software/amazon/documentdb/jdbc/DocumentDbDriver.java

```diff
@@ -69,6 +73,23 @@ public class DocumentDbDriver extends software.amazon.documentdb.jdbc.common.Dri
         new DocumentDbDriver().register();
     }
 
+    private final Supplier<DocumentDbMetadataService> metadataServiceProvider;
+
+    /**
+     * Constructs a new DocumentDbDriver and uses the default metadata service.
+     */
+    public DocumentDbDriver() {
+        metadataServiceProvider = () -> new DocumentDbMetadataServiceImpl();
+    }
+
+    /**
+     * Constructs a new DocumentDbDriver with the provided metadata service.
+     * @param metadataServiceProvider metadata service to use
+     */
+    public DocumentDbDriver(final Supplier<DocumentDbMetadataService> metadataServiceProvider) {
```

Comment:
@normanj-bitquill I'm assuming this will be the entry point to override the schema service for a future local file implementation. Correct?

Reply:
@birschick-bq That is correct. The goal is to supply my own implementation of the DocumentDbMetadataService that will be used for all connections made using the driver.